### PR TITLE
Fix TraceState key validation limits to match W3C spec

### PR DESCRIPTION
**Pull request by priettt**
## Summary

This PR fixes the multi-tenant vendor key validation limits in `ArrayBasedTraceStateBuilder` to properly comply with the W3C Trace Context specification.

**Changes:**
- Update `MAX_VENDOR_ID_SIZE` from 13 to 14 characters
- Update `MAX_TENANT_ID_SIZE` from 240 to 241 characters  
- Update documentation and comments to reflect correct limits
- Add comprehensive test cases for boundary conditions

## Background

The current implementation incorrectly rejects valid keys according to the [W3C Trace Context specification](https://www.w3.org/TR/trace-context/#key). The ABNF rules define:

```
tenant-id = ( lcalpha / DIGIT ) 0*240( lcalpha / DIGIT / "_" / "-"/ "*" / "/" )
system-id = lcalpha 0*13( lcalpha / DIGIT / "_" / "-"/ "*" / "/" )
```

This means:
- `tenant-id`: 1 + 240 = **241 characters maximum**
- `system-id`: 1 + 13 = **14 characters maximum**

## Test Plan

- [x] Added test cases for 241-character tenant IDs (should pass)
- [x] Added test cases for 242-character tenant IDs (should fail)  
- [x] Added test cases for 14-character vendor IDs (should pass)
- [x] Added test cases for 15-character vendor IDs (should fail)
- [x] All existing tests continue to pass
- [x] Ran `./gradlew check` successfully

## Validation

Before this fix:
- Keys with 241-character tenant IDs were incorrectly rejected
- Keys with 14-character vendor IDs were incorrectly rejected

After this fix:
- All test cases pass
- Implementation correctly follows W3C specification

### CODE REVIEW

File: api/all/src/test/java/io/opentelemetry/api/trace/TraceStateTest.java [hunk 1]
```diff
@@ -123,22 +123,39 @@ void testValidAtSignVendorNamePrefix() {
   }
 
   @Test
-  void testVendorIdLongerThan13Characters() {
-    assertThat(TraceState.builder().put("1@nrabcdefghijkl", FIRST_VALUE).build())
+  void testVendorIdWith14Characters() {
+    String key = "1@nrabcdefghijkl";
```
Comment (zeitlinger) [inline api/all/src/test/java/io/opentelemetry/api/trace/TraceStateTest.java:127]: this is 16 chars

Comment (priettt) [reply api/all/src/test/java/io/opentelemetry/api/trace/TraceStateTest.java:127]: Multi-tenant keys are formed with: a `tenant-id`, the "@" symbol, and a `system-id` (called vendor-id in the otel-java codebase).

This test verifies behavior for a `system-id` of 14 chars, so what follows the @ should be that long
